kore/src/Kore/Builtin/List.hs

```diff
@@ -254,11 +251,36 @@ evalGet =
                     (Builtin.appliedFunction . maybeBottom)
                         (Seq.lookup ix _list)
             emptyList <|> bothConcrete
-        )
       where
         maybeBottom =
             maybe Pattern.bottom Pattern.fromTermLike
 
+evalUpdate :: Builtin.Function
+evalUpdate =
+    Builtin.functionEvaluator evalUpdate0
+  where
+    evalUpdate0 :: Builtin.FunctionImplementation
+    evalUpdate0 resultSort [_list, _ix, value] =
+        Builtin.getAttemptedAxiom $ do
+            let emptyList = do
+                    _list <- expectBuiltinList updateKey _list
+                    if Seq.null _list
+                        then Builtin.appliedFunction Pattern.bottom
```

Comment:
Why does this return bottom for an empty list, while returning empty if the index is out of bounds below?